src-docs/src/views/datagrid/schema_columns/control_columns.js

```diff
@@ -314,12 +314,18 @@ export default function DataGrid() {
     pageSize: 15,
   });
   const setPageIndex = useCallback(
-    (pageIndex) => setPagination({ ...pagination, pageIndex }),
-    [pagination, setPagination]
+    (pageIndex) =>
+      setPagination((pagination) => ({ ...pagination, pageIndex })),
+    [setPagination]
```

Comment:
Fun fact: state setters are not required for memoization dependencies so we can make this array empty/fully cached :)
```suggestion
    []
```

Reply:
@tgalfin can you extrapolate this array cleanup to all other callbacks you've cleaned up as part of this PR? thank you for your time!

Reply:
Good callout, addressed here:
[remove unnecessary deps](https://github.com/elastic/eui/pull/6651/commits/34cce090cc92ececf3e31ba425a484fbf1389f8f)